addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -50,9 +71,29 @@ options.registry.TableOfContent = options.Class.extend({
         const $nav = this.$target.find('.s_table_of_content_navbar');
         const $headings = this.$target.find(this.targetedElements);
         $nav.empty();
+        const uniqueHeadingIds = new Set();
         _.each($headings, el => {
             const $el = $(el);
-            const id = 'table_of_content_heading_' + _.now() + '_' + _.uniqueId();
+            if (!this.$target[0].dataset.tocId) {
+                const tocEls = this.$target[0].closest("#wrapwrap").querySelectorAll("section.s_table_of_content");
+                const tocIds = [...tocEls].map(tocEl => parseInt(tocEl.dataset.tocId) || 0);
+                const nextTocId = 1 + Math.max(...tocIds);
+                this.$target[0].dataset.tocId = nextTocId;
+            }
+            if (el.dataset.headingId) {
+                // Reset headingId on duplicate.
+                if (uniqueHeadingIds.has(el.dataset.headingId)) {
+                    delete el.dataset.headingId;
+                } else {
+                    uniqueHeadingIds.add(el.dataset.headingId);
+                }
+            }
+            if (!el.dataset.headingId) {
+                const headingIds = [...$headings].map(headingEl => parseInt(headingEl.dataset.headingId) || 0);
+                const nextHeadingId = 1 + Math.max(...headingIds);
+                el.dataset.headingId = nextHeadingId;
+            }
+            const id = 'table_of_content_heading_' + this.$target[0].dataset.tocId + '_' + el.dataset.headingId;
```

Comment:
Nitpicking: add a small comment on top of this line to explain why it is important to keep an id that does not change too much (why you do this PR).